elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -211,6 +274,51 @@ private static boolean validateSqlInTableConfig(ElideTableConfig elideTableConfi
         return true;
     }
 
+    /**
+     * Validates join clause does not refer to a Table which is not in the same DBConnection.
+     * If joined table is not part of dynamic configuration, then ignore
+     */
+    private static void validateJoinedTablesDBConnectionName(ElideTableConfig elideTableConfig) {
+
+        for (Table table : elideTableConfig.getTables()) {
+            if (!table.getJoins().isEmpty()) {
+
+                Set<String> joinedTables = table.getJoins()
+                        .stream()
+                        .map(join -> join.getTo().toLowerCase(Locale.ENGLISH))
+                        .collect(Collectors.toSet());
+
+                Set<String> connections = elideTableConfig.getTables()
+                        .stream()
+                        .filter(t -> joinedTables.contains(t.getName().toLowerCase(Locale.ENGLISH)))
+                        .map(t -> t.getDbConnectionName())
+                        .collect(Collectors.toSet());
+
+                if (connections.size() > 1 || (connections.size() == 1
+                                && !table.getDbConnectionName().equals(connections.iterator().next()))) {
+                    throw new IllegalStateException("DBConnection name mismatch between table: " + table.getName()
+                                    + " and tables in its Join Clause.");
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates table (or db connection) name is unique across all the dynamic table (or db connection) configs.
+     */
+    private void validateNameUniqueness(Set<? extends Object> configs) {
+
+        Set<String> names = new HashSet<>();
+
+        configs.forEach(obj -> {
+            if (obj instanceof Table && !names.add(((Table) obj).getName().toLowerCase(Locale.ENGLISH))) {
```

Comment:
Maybe Table and DB should both leverage a common interface (Named or something) and use it to get the name without casting here.  This function could take a set of Named instead of a set of objects.

Reply:
Updated